test/support/socket_test_modules.ex

```diff
@@ -16,7 +16,16 @@ end
 defmodule Guardian.Phoenix.SocketTest.MySocket do
   @moduledoc false
   use Phoenix.Socket
-  use Guardian.Phoenix.Socket, module: Guardian.Phoenix.SocketTest.Impl
+  import Guardian.Phoenix.Socket
+  alias Guardian.Phoenix.SocketTest.Impl
+
+  def connect(%{"guardian_token" => token}, socket) do
+    case authenticate(socket, Impl, token) do
+      {:ok, authed_socket} ->
```

Comment:
@scrogson do you happen to know if there is any extra overhead to one of these vs the other:

Making a new tuple:

```elixir
  def connect(%{"guardian_token" => token}, socket) do
    case authenticate(socket, Impl, token) do
      {:ok, authed_socket} -> {:ok, authed_socket}
      _ -> :error
    end
  end
```

Capturing and returning the capture:

```elixir
  def connect(%{"guardian_token" => token}, socket) do
    case authenticate(socket, Impl, token) do
      {:ok, authed_socket} = result -> result
      _ -> :error
    end
  end
```

Reply:
I don't know for sure. It certainly would be too negligible to make a difference. From a code style perspective, I prefer:

```ex
def connect(%{"token" => token}, socket) do
  case authenticate(socket, Impl, token) do
    {:ok, socket} ->
      {:ok, socket}
    _ ->
      :error
  end
end
```

Reply:
or

```ex
case authenticate(socket, Impl, token) do
  {:ok, socket} -> {:ok, socket}
  {:error, _error} -> :error
end
```

Reply:
Coolio!  I'm a fan of fewer newlines myself.  I was just curious about the capture vs create, seen both around 😁